api/utils.py

```diff
@@ -218,23 +226,72 @@ def addWorkMeta(cls, work, **kwargs):
 
     @classmethod
     def formatEditionOutput(
-        cls, edition, records=None, showAll=False, formats=None, reader=None
+        cls, edition, records=None, dbClient=None, showAll=False, formats=None, reader=None
     ):
         editionWorkTitle = edition.work.title
         editionWorkAuthors = edition.work.authors
+        editionInCollection = cls.checkEditionInCollection(None, edition, dbClient)
         
         return cls.formatEdition(
-            edition, editionWorkTitle, editionWorkAuthors, records, formats, showAll=showAll, reader=reader
+            edition, editionWorkTitle, editionWorkAuthors, editionInCollection, records, formats, showAll=showAll, reader=reader
         )
-    
+
+    @classmethod
+    def checkEditionInCollection(cls, work, edition, dbClient):
+
+        collectionIDs = []
+
+        if work != None:
+            for edit in dbClient.session.query(Edition) \
+                .filter(Edition.work_id == work.id):
+                    for collectEdit in dbClient.session.query(COLLECTION_EDITIONS) \
+                        .filter(COLLECTION_EDITIONS.c.edition_id == edit.id):
```

Comment:
I'm a little concerned about the depth of the nested `for` loops here since it could spawn many database queries that would slow down our response times. I'm fairly certain it should be able to turn this into a single query which would help with that, but let me know if there are limitations on that approach.

Reply:
I'll attempt to turn this into a simplified query.